modules/phase_field/include/materials/GammaAndLAnisoGauss.h

```diff
@@ -0,0 +1,151 @@
+// 5D Gaussian anisotropy material object
+// Material properties adding anisotropy to gamma and L.
+
+#pragma once
+
+#include "ADMaterial.h"
+#include "MooseMesh.h"
+#include "MathUtils.h"
+#include "GrainTrackerInterface.h"
+#include <vector>
+
+class GammaAndLAnisoGauss : public ADMaterial
+{
+public:
+  static InputParameters validParams();
+  GammaAndLAnisoGauss(const InputParameters & parameters);
+
+protected:
+  virtual void computeQpProperties() override;
+
+private:
+  void readQuaternionFile();
+  void readLibraryFile();
+
+  // Grains orientation quaternions.
+  std::vector<double> _qwR, _qxR, _qyR, _qzR;
+  // Normalized_Axis_X  Normalized_Axis_Y Normalized_Axis_Z  Angle(Radian) GB_Normal_X  GB_Normal_Y
+  // GB_Normal_Z  Minimun_GB_energy(J/m^2) Minimun_L(nm^3/eV.ns).
+  std::vector<double> _vbalibx, _vbaliby, _vbalibz, _thetabalib, _miubalibx, _miubaliby, _miubalibz,
+      _MinimaEnergy, _MinimaL;
```

Comment:
and again

Reply:
I ended merging all four materials into a single file